runtime/src/main/cpp/MemorySharedRefs.hpp

```diff
@@ -10,14 +10,32 @@
 
 #include "Memory.h"
 
+// TODO: Generalize for uses outside this file.
+enum class ErrorHandlingPolicy {
+  kIgnore,
+  kReturnDefault,
+  kThrowException,
+  kTerminate,
+};
+
 class KRefSharedHolder {
  public:
   void initLocal(ObjHeader* obj);
 
   void init(ObjHeader* obj);
 
+  // Error if called from the wrong worker with non-frozen obj_.
+  template <ErrorHandlingPolicy errorHandlingPolicy>
   ObjHeader* ref() const;
-  ObjHeader* refOrNull() const;
+  ObjHeader* refOrTerminate() const {
```

Comment:
What is the value of those wrappers? It is essentially a parameter value hardcoded to a function name. Looks like anti-pattern, isn't it? There are more below.

Reply:
I personally find them more readable. But I'm not overly attached to them, so they can go once enum is renamed to be a tad shorter.